linker/shared/src/test/scala/org/scalajs/linker/AnalyzerTest.scala

```diff
@@ -593,6 +593,40 @@ class AnalyzerTest {
     }
   }
 
+  @Test
+  def jmBigNumbersNotInstantiatedWhenUsingStringFormat(): AsyncResult = await {
```

Comment:
Hmmm... I don't feel this test is in the right spot. I realize that we have precedent with `juPropertiesNotReachableWhenUsingGetSetClearProperty` so probably we should leave it in this PR and then move it out.

Basically what is happening here is that we mix test that:
- Test the analyzer implementation itself.
- Test the DCE properties of a specific library implementation.

So I think we should separate these. (it also seems like there are some commonalities between this test and the juProps test that might be worth factoring out in such a new test class).